frontend/ui/widget/virtualkeyboard.lua

```diff
@@ -571,8 +572,38 @@ function VirtualKeyboard:init()
     end
 end
 
-function VirtualKeyboard:getKeyboardLayout()
-    return G_reader_settings:readSetting("keyboard_layout") or G_reader_settings:readSetting("language")
+function VirtualKeyboard:getKeyboardLocale()
```

Comment:
I wouldn't change this function name. Not only is "keyboard layout" a standard term, but it's safely decoupled from locale-related shenanigans. Instead this calls up all kinds of locale-related questions and doubts.

Reply:
as i commented below i am trying to separate language from layout, as calling it a layout is more confusing to me.

if `Locale` is confusing i can change it to `getKeyboardLanguage` which is more definitive.

Reply:
>  i am trying to separate language from layout

Which is precisely why it should be layout — not locale and especially not language.

Reply:
the way i see it is that the 8 or 12 ways in each language are layouts, there are many languages and each language has different layouts. if you want that languages should be called layouts than the current layouts should be called something like keymaps